numpy/core/tests/test_umath.py

```diff
@@ -456,7 +456,10 @@ def assert_complex_equal(x, y):
 
     def test_fast_power(self):
         x = np.array([1, 2, 3], np.int16)
-        assert_((x**2.00001).dtype is (x**2.0).dtype)
+        res = x**2.0
+        assert_((x**2.00001).dtype is res.dtype)
+        assert_(res is not x)
```

Comment:
I think I'd like a `not np.shares_memory(res, x)` here too - far too easy to break identity while still having a view.

Also a check that `x` is still `[1, 2, 3]` would do the job